### PR TITLE
Feature offlineforms

### PR DESCRIPTION
**Pull request by micahmills**
There is still some styling and UI tweaks that need to be done, but the offline capability is working.

### CODE REVIEW

**Comment by corsacca** [comment]
@micahmills is this PR ready to be merged?

**Comment by micahmills** [comment]
I don't remember. I will have to look at it again.

**Comment by micahmills** [comment]
@corsacca I believe this is ready to go now.

**Comment by corsacca** [comment]
Fields are reflecting customization:
image

File: public/public.js [hunk 1]
```diff
@@ -102,20 +213,23 @@ jQuery(document).ready(function () {
...
+    //     button.html( window.TRANSLATION.submit_in + ' ' + counter)
+    //     --counter;
+
+    //     if ( counter === 0 ) {
+    //         clearInterval(myInterval);
+    //         button.html( window.TRANSLATION.submit ).prop('disabled', false)
+    //     }
 
-    }, 1000);
+    // }, 1000);
```
Comment (corsacca) [inline public/public.js:232]: comments on this section?

Comment (micahmills) [reply public/public.js]: This should have been uncommented. I turned it off during testing, must have forgot to turn it back on.

**Comment by micahmills** [comment]
The fields not reflecting changes is because we are caching the file with a service worker. I have changed the Service worker so that it will serve from the cache but then update the cache if there is a network connection. This means that if you make a change you will need to refresh the site twice, once to refresh the cache and the second time to show that cached version.

**Comment by micahmills** [comment]
The Travis CI issues appear to be inherited from merging from master.

**Comment by ChrisChasm** [comment]
I'm going to pull this Request, clean up the phpcs, and then pull it into the v3 effort.

**Pull request merged**